src/nc/input/mach-o/MachOParser.cpp

```diff
@@ -290,9 +326,58 @@ class MachOParserImpl {
                 }
             }
 
-            image_->addSymbol(std::make_unique<core::image::Symbol>(type, name, value, section));
+            auto sym = std::make_unique<core::image::Symbol>(type, name, value, section);
+            symbols_.push_back(sym.get());
+            image_->addSymbol(std::move(sym));
+        }
+    }
+
+    template<class Mach>
+    void parseDySymtabCommand() {
+        dysymtab_command command;
+        if (!read(source_, command)) {
+            throw ParseError(tr("Could not read dsymtab command."));
+        }
+        byteOrder_.convertFrom(command.ilocalsym);
+        byteOrder_.convertFrom(command.nlocalsym);
+        byteOrder_.convertFrom(command.iextdefsym);
+        byteOrder_.convertFrom(command.nextdefsym);
+        byteOrder_.convertFrom(command.iundefsym);
+        byteOrder_.convertFrom(command.nundefsym);
+        byteOrder_.convertFrom(command.indirectsymoff);
+        byteOrder_.convertFrom(command.nindirectsyms);
+
+        if (!source_->seek(command.indirectsymoff)) {
+            throw ParseError(tr("Could not seek to the string table."));
+        }
+
+        foreach (const auto &indirectSection, indirectSections_) {
+            if (!source_->seek(command.indirectsymoff + indirectSection.index * sizeof(uint32_t))) {
+                throw ParseError(tr("Could not seek to the string table."));
+            }
+
+            for (uint32_t i = 0; i < indirectSection.size; i += indirectSection.stride) {
+                using core::image::Symbol;
+                using core::image::SymbolType;
+
+                uint32_t symbolIndex;
+
+                if (!read(source_, symbolIndex)) {
+                    throw ParseError(tr("Could not read symbol number %1.").arg(i));
+                }
+                byteOrder_.convertFrom(symbolIndex);
+
+                if (! (symbolIndex & INDIRECT_SYMBOL_LOCAL || symbolIndex & INDIRECT_SYMBOL_ABS)) {
+                    if (symbolIndex >= symbols_.size()) {
+                        throw ParseError(tr("symbol index too large %1.").arg(symbolIndex));
+                    }
+                    const core::image::Symbol * sym = symbols_[symbolIndex];
+                    image_->addSymbol(std::make_unique<core::image::Symbol>(sym->type(), sym->name(), indirectSection.addr + i, indirectSection.section));
```

Comment:
Two files: an input (a Mach-O executable with relocations) and a correct output for it.
See https://github.com/yegord/snowman/blob/master/doc/tests.asciidoc